src/log4stash/Filters/JsonFilter.cs

```diff
@@ -76,7 +76,7 @@ private static void ScanToken(IDictionary<string, object> logEvent, JToken token
 
         private static string Join(string prefix, string name)
         {
-            return (string.IsNullOrEmpty(prefix) ? name : prefix + "." + name);
+            return (string.IsNullOrEmpty(prefix) ? name : prefix + "_" + name);

```

Comment:
The dot is not allowed in field names since elastic 2.0...
[(https://www.elastic.co/blog/introducing-the-de_dot-filter]
)"One of these changes was that Elasticsearch 2.0 does not support field names with a . (or dot character) in them"